### DISCUSSION TITLE
Using openapi with reverse proxies

### DISCUSSION THREAD

**Discussion by jessevitrone** [PostgREST/postgrest#3773] [Q&A] [answered]
I'd like to use openapi and swagger, but also will probably need to use some kind of reverse proxy for URIs.  Do these two features work well together?  It seems like using a reverse proxy would result in different URI's than the ones generated by openapi.  Is there a good way to make these work together?

**Comment by laurenceisla** [comment]
> Is there a good way to make these work together?

Yes! You can set your base URL for the OpenAPI output in the [`openapi-server-proxy-uri`](https://docs.postgrest.org/en/stable/references/configuration.html#openapi-server-proxy-uri) configuration variable.

**Reply by jessevitrone** [reply]
Thanks for the suggestion!  What if it's more than a base url change?  Like changing a view that is `mlb_teams` to `/mlb/teams` or `/mlb-teams`?  These are simple examples but I'm thinking in general about formatting changes where we don't want the URIs to be necessarily tied one-to-one with the table or view names.

Reverse proxies were the only solution that I saw for customizing the URIs / endpoints, but since that happens before the request comes in to postgrest, it seems like the generated openapi wouldn't know about it.

**Reply by laurenceisla** [reply, marked as answer]
> formatting changes where we don't want the URIs to be necessarily tied one-to-one with the table or view names [...] but since that happens before the request comes in to postgrest, it seems like the generated openapi wouldn't know about it.

Ah, you're right there. For changes where the proxy endpoints are different from what PostgREST builds, then the OpenAPI won't generate those same endpoints.

So yeah, for now I can't see a workaround for those cases, except perhaps naming the views themselves with the special characters (not sure if there could be extra problems there, e.g. #3771). This could be possible to do with the upcoming [`postgrest-openapi`](https://github.com/PostgREST/postgrest-openapi) project.

**Reply by jessevitrone** [reply]
That's what I suspected, thanks for the confirmation
